components/director/internal/domain/runtime/repository.go

```diff
@@ -2,148 +2,98 @@ package runtime
 
 import (
 	"context"
-	"database/sql"
 	"fmt"
 
 	"github.com/google/uuid"
-
+	"github.com/kyma-incubator/compass/components/director/internal/domain/label"
+	"github.com/kyma-incubator/compass/components/director/internal/repo"
 	"github.com/pkg/errors"
 
-	"github.com/kyma-incubator/compass/components/director/internal/domain/label"
 	"github.com/kyma-incubator/compass/components/director/internal/labelfilter"
 	"github.com/kyma-incubator/compass/components/director/internal/model"
 	"github.com/kyma-incubator/compass/components/director/internal/persistence"
-	"github.com/kyma-incubator/compass/components/director/pkg/pagination"
 	"github.com/lib/pq"
 )
 
 const runtimeTable string = `"public"."runtimes"`
 
 const runtimeFields string = `"id", "tenant_id", "name", "description", "status_condition", "status_timestamp", "auth"`
 
-type pgRepository struct{}
+type pgRepository struct {
+	existQuerier *repo.ExistQuerier
+	*repo.SingleGetter
+	*repo.Deleter
+	*repo.PageableQuerier
+}
 
 func NewPostgresRepository() *pgRepository {
-	return &pgRepository{}
+	return &pgRepository{
+		existQuerier:    repo.NewExistQuerier(runtimeTable, "tenant_id"),
+		SingleGetter:    repo.NewSingleGetter(runtimeTable, "tenant_id", "id", runtimeFields),
+		Deleter:         repo.NewDeleter(runtimeTable, "tenant_id", "id"),
+		PageableQuerier: repo.NewPageableQuerier(runtimeTable, "tenant_id", runtimeFields),
+	}
 }
 
 func (r *pgRepository) Exists(ctx context.Context, tenant, id string) (bool, error) {
-	persist, err := persistence.FromCtx(ctx)
-	if err != nil {
-		return false, errors.Wrap(err, "while fetching DB from context")
-	}
-
-	stmt := fmt.Sprintf(`SELECT 1 FROM %s WHERE "id" = $1 AND "tenant_id" = $2`,
-		runtimeTable)
-
-	var count int
-	err = persist.Get(&count, stmt, id, tenant)
-	if err != nil {
-		if err == sql.ErrNoRows {
-			return false, nil
-		}
-		return false, errors.Wrap(err, "while getting runtime from DB")
-	}
-
-	return true, nil
+	return r.existQuerier.Exists(ctx, tenant, repo.Conditions{{ID: "id", Val: id}})
```

Comment:
This approach looks really good
![tenor-75024046](https://user-images.githubusercontent.com/7155799/63017123-c66dcd00-be95-11e9-8f05-708136549a51.gif)